dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/deduplication/DeduplicationHelper.java

```diff
@@ -87,12 +89,22 @@ public String getInvalidReferenceErrors( DeduplicationMergeParams params )
             }
         }
 
-        for ( String rel : mergeObject.getRelationships() )
+        // Check if any references not on duplicate is used
+        String rel = mergeObject.getRelationships().stream()
+            .filter( r -> !validRelationships.contains( r ) )
+            .findFirst()
+            .orElse( null );
+
+        if ( rel != null )
         {
-            if ( original.getUid().equals( rel ) || !validRelationships.contains( rel ) )
-            {
-                return "Invalid relationship '" + rel + "'";
-            }
+            return "Invalid relationship '" + rel + "'. Refers to itself.";
+        }
+
+        String error = getDuplicateRelationshipError( params.getOriginal(), params.getDuplicate() );
```

Comment:
seems to me we are checking self-referencing relationship ... can we change the method name for readability?

Reply:
We actually check for relationships where either to or from are the same; So we also deny now block for relationships that would result in the same (IE. A->B + C->B would be the same)